lib/lookup/src/lookup_v2/owned.rs

```diff
@@ -39,6 +44,50 @@ impl OwnedPath {
         vec![OwnedSegment::field(field)].into()
     }
 
+    /// Create the possible fields that can be followed by this lookup.
+    /// Because of coalesced paths there can be a number of different combinations.
+    /// There is the potential for this function to create a vast number of different
+    /// combinations if there are multiple coalesced segments in a path.
+    ///
+    /// The limit specifies the limit of the path depth we are interested in.
+    /// Metrics is only interested in fields that are up to 3 levels deep (2 levels + 1 to check it
+    /// terminates).
```

Comment:
This entire function was moved, so the comments aren't new (it was just switched to the new path / lookup code). However, I believe it just means it can check if you reached the last query segment in a path.